### ISSUE TITLE
new rule - `isinstance` on class that has an unsafe overlap with a `@runtime_checkable` `Protocol`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3760]
https://pyrefly.org/en/docs/error-kinds/#unsafe-overlap

```py
from typing import Protocol, runtime_checkable

@runtime_checkable
class P(Protocol):
    x: int

class C:
    x: str

c = C()
if isinstance(c, P):  # should be an error
    pass
```

[pyright also detects this](https://basedpyright.com/?pythonVersion=3.15&typeCheckingMode=all&reportUninitializedInstanceVariable=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYA0UIArijFgKYD6pAFs6QNYCGAIwrMAUCIAC9Ri3Zdeg4SNIU%2BAZ1WEAFETAlyFAJQAuEVDNQAHkcyMxytRoDCJ85euqYIO1AC8UR5oGIkjAmKqoHnwopMyapDQExqbmCA5AA)
